corehq/apps/users/audit/change_messages.py

```diff
@@ -1,91 +1,329 @@
+    change: either 'set' / 'reset' / 'remove' / 'add'
+    change_details: a Boolean or
+                    a dict with details as needed for the change
+                    a list of a dicts as needed for the change
+    """
     @staticmethod
     def program_change(program):
         if program:
-            message = f"Program: {program.name}[{program.get_id}]"
+            change_message = {"program": {Change.SET: {"id": program.get_id, "name": program.name}}}
         else:
-            message = 'Program: None'
-        return message
+            change_message = {"program": {Change.SET: {}}}
+        return change_message
 
     @staticmethod
     def role_change(user_role):
-        message = 'Role: None'
         if user_role:
-            message = f"Role: {user_role.name}[{user_role.get_qualified_id()}]"
-        return message
+            change_message = {"role": {Change.SET: {"id": user_role.get_qualified_id(), "name": user_role.name}}}
+        else:
+            change_message = {"role": {Change.SET: {}}}
+        return change_message
 
     @staticmethod
     def domain_removal(domain):
-        return f"Removed from domain '{domain}'"
+        return {"domain": {Change.REMOVE: {"name": domain}}}
 
     @staticmethod
     def registered_devices_reset():
-        return "Registered devices reset"
+        return {"devices": {Change.RESET: True}}
 
     @staticmethod
     def two_factor_disabled_for_days(days):
-        return f"Disabled for {days} days"
+        return {"two_factor": {Change.REMOVE: {"days": days}}}
 
     @staticmethod
     def two_factor_disabled_with_verification(verified_by, verification_mode):
-        return f'Two factor disabled. Verified by: {verified_by}, verification mode: "{verification_mode}"'
+        return {
+            "two_factor": {
+                Change.REMOVE: {
+                    "verified_by": verified_by,
+                    "verification_mode": verification_mode
+                }
+            }
+        }
 
     @staticmethod
     def password_reset():
-        return "Password reset"
+        return {"password": {Change.RESET: True}}
 
     @staticmethod
-    def status_update(action, reason):
-        return f'User {action}. Reason: "{reason}"'
+    def status_update(active, reason):
+        return {
+            "status": {
+                Change.SET: {
+                    "active": active,
+                    "reason": reason
+                }
+            }
+        }
 
     @staticmethod
-    def phone_number_added(phone_number):
-        return f"Added phone number {phone_number}"
+    def phone_numbers_added(phone_numbers):
+        return {
+            "phone_numbers": {Change.ADD: phone_numbers}
+        }
 
     @staticmethod
-    def phone_number_removed(phone_number):
-        return f"Removed phone number {phone_number}"
+    def phone_numbers_removed(phone_numbers):
+        return {
+            "phone_numbers": {Change.REMOVE: phone_numbers}
```

Comment:
what is `phone_numbers`? Is it a list of strings?

Reply:
Ya. List of phone numbers